git_machete/github.py

```diff
@@ -24,9 +25,35 @@ def __repr__(self) -> str:
 GITHUB_TOKEN_ENV_VAR = 'GITHUB_TOKEN'
 
 
-def github_token() -> Optional[str]:
+def _token_from_gh() -> Optional[str]:
+    proc = subprocess.run(
+        ["gh", "auth", "status", "--show-token"],
```

Comment:
Actually... what happens if/should we specifically care about:
1. `gh` not being available in the system (very likely),
2. `gh auth status --show-token` timing out for whatever reason (probably not so likely?)
❓

Reply:
1. resolved
2. `gh auth status` gets the token configuration from the filesystem, not an external call, so the command is non-blocking.